app/services/puppet_fact_parser.rb

```diff
@@ -140,6 +140,17 @@ def cores
     facts.dig('processors', 'count') || facts['processorcount']
   end
 
+  def total_disk_size
+    disks      = facts.select { |key, _| key.to_s =~ /blockdevice.*_size/ }
```

Comment:
I think this is confusing. The method is named `total_disk_size` but here you look at partitions. It can also be confusing if you have loop devices. Facter 3 has a disks fact:
```console
$ facter --json disks
{
  "disks": {
    "vda": {
      "size": "50.00 GiB",
      "size_bytes": 53687091200,
      "vendor": "0x1af4"
    }
  }
}
```

Perhaps that's more reliable?

Reply:
Yes this is for historical reasons, discovery did not support facter3 facts. I think it should be fine to have this robust implementation for F3 and a simple fallback implementation for F2 just to show something.